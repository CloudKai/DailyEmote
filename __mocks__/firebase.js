const firebase = jest.requireActual('firebase/app');
require('firebase/firestore');

const mockFirebase = {
  firestore: jest.fn(() => ({
    collection: jest.fn().mockReturnThis(),
    doc: jest.fn().mockReturnThis(),
    get: jest.fn().mockResolvedValue({
      exists: true,
      data: () => ({
        avatar: 'http://test.com/photo.jpg',
        username: 'Test User',
        email: 'test@example.com',
        password: 'password123',
      }),
    }),
    set: jest.fn().mockResolvedValue(undefined),
    update: jest.fn().mockResolvedValue(undefined),
  })),
};

firebase.initializeApp = jest.fn(() => mockFirebase);
module.exports = firebase;
