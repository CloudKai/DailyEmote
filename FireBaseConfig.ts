// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAuth, getReactNativePersistence } from "firebase/auth";
import ReactNativeAsyncStorage from '@react-native-async-storage/async-storage';
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyCZpRx2Yfq4qx68w7qkFD9h5072AR5p-E4",
  authDomain: "newproject-87d28.firebaseapp.com",
  projectId: "newproject-87d28",
  storageBucket: "newproject-87d28.appspot.com",
  messagingSenderId: "1067322831926",
  appId: "1:1067322831926:web:df158d2620c399721b5497",
  measurementId: "G-H6K6834BF1"
};

// Initialize Firebase
export const FIREBASE_APP = initializeApp(firebaseConfig);
export const FIREBASE_AUTH = getAuth(FIREBASE_APP);