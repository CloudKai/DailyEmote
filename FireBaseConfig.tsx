// Import the functions you need from the SDKs you need
import *as firebase from "firebase/app";
import { getApp, getApps, initializeApp } from "firebase/app";
import { getAuth, initializeAuth, getReactNativePersistence } from "firebase/auth";
import { connectFirestoreEmulator, getFirestore } from 'firebase/firestore';
import AsyncStorage from "@react-native-async-storage/async-storage";

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyCZpRx2Yfq4qx68w7qkFD9h5072AR5p-E4",
  authDomain: "newproject-87d28.firebaseapp.com",
  projectId: "newproject-87d28",
  storageBucket: "newproject-87d28.appspot.com",
  messagingSenderId: "1067322831926",
  appId: "1:1067322831926:web:3084ff3a61db51971b5497",
  measurementId: "G-Z9Z1ZNGTF9"
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
initializeAuth(app, {
  persistence: getReactNativePersistence(AsyncStorage),
});
export const FIREBASE_AUTH = getAuth(app);