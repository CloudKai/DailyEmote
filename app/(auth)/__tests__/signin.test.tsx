//Not working!
import React from 'react'
import SignInPage from '../signin';
import '@testing-library/jest-dom';
import { FIREBASE_AUTH } from '../../../FireBaseConfig';
import { signInWithEmailAndPassword, signOut, onAuthStateChanged, initializeAuth, getIdToken } from 'firebase/auth';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Alert } from 'react-native';
//Install these 2 dependancies then install expo again to work


jest.spyOn(Alert, 'alert');

it('Check all elements are present', () => {
  const page = render(<SignInPage />);
  // Snapshot of the page
  // expect(tree).toMatchSnapshot();

  //Image Logo is present
  expect(page.getByTestId('logo'));

  //TextBoxes w the right placeholder are present
  expect(page.getByTestId("emailTextBox").props.placeholder).toEqual("Email");
  expect(page.getByTestId("pwdTextBox").props.placeholder).toEqual("Password");

  //Buttons are present
  expect(page.getByTestId("loginButton"));
  expect(page.getByTestId("signUpButton"));
})


// it('Show all errors when sign in failed', async () => {
//   const page = render(<SignInPage />);

//   const loginButton = page.getByTestId("loginButton");
//   const emailTextInput = page.getByTestId("emailTextBox");
//   const pwdTextInput = page.getByTestId("pwdTextBox");

//   //Empty Fields
//   fireEvent.changeText(emailTextInput, "");
//   fireEvent.changeText(pwdTextInput, "");
//   await fireEvent.press(loginButton);
//   expect(Alert.alert).toHaveBeenCalledWith('Invalid Email', 'Enter a valid email');
//   // fireEvent.press(Alert.prompt[0]);

//   //Invalid Email
//   fireEvent.changeText(emailTextInput, "kai");
//   fireEvent.changeText(pwdTextInput, "");
//   await fireEvent.press(loginButton);
//   expect(Alert.alert).toHaveBeenCalledWith('Invalid Email', 'Enter a valid email');
  

//   //Valid Email but Empty Pwd Fields
//   fireEvent.changeText(emailTextInput, "kai@gmail.com");
//   fireEvent.changeText(pwdTextInput, "");
//   await fireEvent.press(loginButton);
//   expect(Alert.alert).toHaveBeenCalledWith('Empty Password Fields', 'Enter your password');
  

//   //Valid Email but Invalid Pwd 
//   fireEvent.changeText(emailTextInput, "kai@gmail.com");
//   fireEvent.changeText(pwdTextInput, "123"); //Incorrect Password for email
//   await fireEvent.press(loginButton);
//   expect(Alert.alert).toHaveBeenCalledWith('Invalid credentials', 'Wrong credentials');

// })


it('Able to go to sign up page', () => {
  const page = render(<SignInPage />);

})

it('Sign in Successfully', async () => {
  const page = render(<SignInPage />);

  const loginButton = page.getByTestId("loginButton");
  const emailTextInput = page.getByTestId("emailTextBox");
  const pwdTextInput = page.getByTestId("pwdTextBox");

  const email = "kaim@gmail.com";
  const password = "kai123";

  fireEvent.changeText(emailTextInput, email);
  fireEvent.changeText(pwdTextInput, password);
  await fireEvent.press(loginButton);
  // const user = await signInWithEmailAndPassword(FIREBASE_AUTH, email, password);
  // expect(user.user).toBeTruthy();
  // expect(FIREBASE_AUTH.currentUser).not.toBe(null);

  // expect(page).toMatchSnapshot();
})

