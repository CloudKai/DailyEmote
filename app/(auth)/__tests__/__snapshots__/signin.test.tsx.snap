// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in Successfully 1`] = `[Function]`;
